lib/pbench/server/database/models/api_keys.py

```diff
@@ -66,9 +66,13 @@ def add(self):
         except Exception as e:
             Database.db_session.rollback()
             self.logger.error("Can't add {} to DB: {}", str(self), str(e))
-            raise decode_integrity_error(
+            decode_exc = decode_integrity_error(
                 e, on_duplicate=DuplicateApiKey, on_null=NullKey
             )
+            if decode_exc == e:
```

Comment:
This should probably be `if decode_exc is e` ... that is, it's not just `Object.__eq__(e)` (whatever logic the superclass `__eq__` chooses, which might not be much) but the exact same exception instance. Although in the context of this logic I'm not sure it makes a functional difference since the other paths aren't likely to return an equal or identical instance. 😆